plugins/janus_videoroom.c

```diff
@@ -2208,6 +2189,68 @@ void janus_videoroom_slow_link(janus_plugin_session *handle, int uplink, int vid
 	}
 }
 
+static void janus_videoroom_recorder_create(janus_videoroom_participant *participant, gboolean audio, gboolean video) {
```

Comment:
I guess that both `recorder_create` and `recorder_close` could lock the `rec_mutex` themselves. Or would you rather leave this to the caller?